bundles/scripting/core/src/main/java/org/apache/sling/scripting/core/impl/DefaultSlingScript.java

```diff
@@ -371,21 +380,34 @@ public void setReader(Reader reader) {
             }
 
             // evaluate the script
-            final Object result;
-            if (method == null && this.scriptEngine instanceof Compilable) {
-                CachedScript cachedScript = scriptCache.getScript(scriptName);
-                if (cachedScript == null) {
-                    ScriptNameAwareReader snReader = new ScriptNameAwareReader(reader, scriptName);
-                    CompiledScript compiledScript = ((Compilable) scriptEngine).compile(snReader);
-                    cachedScript = new CachedScriptImpl(scriptName, compiledScript);
-                    scriptCache.putScript(cachedScript);
-                    LOGGER.debug("Adding {} to the script cache.", scriptName);
+            Object result = null;
+            String selectorString = props.getRequest().getRequestPathInfo().getSelectorString();
+            if(StringUtils.isBlank(selectorString) || !selectorString.contains(BVP_VARIABLES_SELECTOR)) {
+                LOGGER.debug("No " + BVP_VARIABLES_SELECTOR + " selector found in the request URI. Script will be evaluated.");
+                if (method == null && this.scriptEngine instanceof Compilable) {
+                    CachedScript cachedScript = scriptCache.getScript(scriptName);
+                    if (cachedScript == null) {
+                        ScriptNameAwareReader snReader = new ScriptNameAwareReader(reader, scriptName);
+                        CompiledScript compiledScript = ((Compilable) scriptEngine).compile(snReader);
+                        cachedScript = new CachedScriptImpl(scriptName, compiledScript);
+                        scriptCache.putScript(cachedScript);
+                        LOGGER.debug("Adding {} to the script cache.", scriptName);
+                    } else {
+                        LOGGER.debug("Script {} was already cached.", scriptName);
+                    }
+                    result = cachedScript.getCompiledScript().eval(ctx);
                 } else {
-                    LOGGER.debug("Script {} was already cached.", scriptName);
+                    result = scriptEngine.eval(reader, ctx);
                 }
-                result = cachedScript.getCompiledScript().eval(ctx);
             } else {
-                result = scriptEngine.eval(reader, ctx);
+                LOGGER.warn("Found " + BVP_VARIABLES_SELECTOR + " selector in the request URI. "
+                        + "Script will not be evaluated. Returning available scripting variables as json...");
+                acquireScriptingVariables(ctx, bvpVariables, 100, scriptEngine.getFactory().getEngineName());
```

Comment:
The according constants from ScriptContext (i.e. ENGINE_SCOPE or GLOBAL_SCOPE) should be used instead of the magic integers